### PR TITLE
log-backup: update for compression type compatibility and config default value update

### LINKED ISSUE #13456: log-backup: compatibility for different compression type
## Development Task
- [x] compatibility for different compression type
- config default value
- - [x] log-backup.enable = true
- - [x] log-backup.max-flush-interval = 3min

### PR DESCRIPTION
**Pull request by Leavrth**
Signed-off-by: Leavrth <jianjun.liao@outlook.com>

### What is changed and how it works?

Issue Number: Close #13456 

What's Changed: update for compression type compatibility.

```commit-message
```

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
  [test record](https://pingcap.feishu.cn/docx/doxcn1HG0B3lDZq1Ni5l2j5l5Sh)

### Release note 

```release-note
None.
```

### CODE REVIEW

**Comment by joccau** [comment]
Manual test
https://pingcap.feishu.cn/docx/doxcn1HG0B3lDZq1Ni5l2j5l5Sh

**Review by joccau — APPROVED** [review]
LGTM

File: components/backup-stream/src/utils.rs [hunk 1]
```diff
@@ -627,6 +632,81 @@ impl AsyncRead for FilesReader {
     }
 }
 
+#[async_trait::async_trait]
+pub trait CompressionWriter: Sync + Send + 'static {
```
Comment (YuJuncen) [inline components/backup-stream/src/utils.rs:636]: Can we make it a sub trait of `AsyncWrite`?

File: components/external_storage/export/src/export.rs [hunk 2]
```diff
@@ -333,17 +333,36 @@ impl ExternalStorage for EncryptedExternalStorage {
...
-        } else {
-            self.read(storage_name)
+        let reader = {
+            let r = if let Some((off, len)) = range {
+                self.read_part(storage_name, off, len)
+            } else {
+                self.read(storage_name)
+            };
+
+            match compression_type {
```
Comment (YuJuncen) [inline components/external_storage/export/src/export.rs:350]: Can we extract the logic here out to something like `compression_writer_dispatcher`?

File: components/external_storage/src/lib.rs [hunk 3]
```diff
@@ -85,17 +86,36 @@ pub trait ExternalStorage: 'static + Send + Sync {
...
+        let reader = {
+            let r = if let Some((off, len)) = range {
+                self.read_part(storage_name, off, len)
+            } else {
+                self.read(storage_name)
+            };
+
+            match compression_type {
+                Some(c) => match c {
+                    CompressionType::Unknown => r,
```
Comment (overvenus) [inline components/external_storage/src/lib.rs:105]: Please add some comments.

File: components/external_storage/export/src/export.rs [hunk 4]
```diff
@@ -333,17 +333,36 @@ impl ExternalStorage for EncryptedExternalStorage {
...
+                Some(c) => match c {
+                    CompressionType::Unknown => r,
```
Comment (overvenus) [inline components/external_storage/export/src/export.rs:352]: Please add some comments.

File: components/backup-stream/src/utils.rs [hunk 5]
```diff
@@ -627,6 +632,81 @@ impl AsyncRead for FilesReader {
...
+    async fn write_all(&mut self, src: &[u8]) -> Result<()>;
+
+    async fn shutdown(&mut self) -> Result<()>;
```
Comment (overvenus) [inline components/backup-stream/src/utils.rs:639]: What does `shutdown` do?

File: components/external_storage/export/src/export.rs [hunk 6]
```diff
@@ -333,17 +333,36 @@ impl ExternalStorage for EncryptedExternalStorage {
         &self,
         storage_name: &str,
         restore_name: std::path::PathBuf,
-        compressed_range: Option<(u64, u64)>,
+        range: Option<(u64, u64)>,
```
Comment (YuJuncen) [inline components/external_storage/export/src/export.rs:336]: There are too many arguments to the function, can we pack them into something like `RestoreConfig`?

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

File: components/backup-stream/src/utils.rs (hunk 1 above)
Comment (YuJuncen) [reply components/backup-stream/src/utils.rs:636]: Making it a subtrait of `AsyncWrite` can eliminate a boxed future, and I think `write_all` would be a hot function.

File: components/backup-stream/src/utils.rs (hunk 5 above)
Comment (Leavrth) [reply components/backup-stream/src/utils.rs:639]: flush the data in memory into the file.
`shutdown` means close the write stream.

Comment (overvenus) [reply components/backup-stream/src/utils.rs:639]: Please add comments to these two methods.

File: components/external_storage/src/lib.rs (hunk 3 above)
Comment (Leavrth) [reply components/external_storage/src/lib.rs]: done

File: components/external_storage/export/src/export.rs (hunk 4 above)
Comment (Leavrth) [reply components/external_storage/export/src/export.rs]: done

File: components/backup-stream/src/utils.rs (hunk 5 above)
Comment (Leavrth) [reply components/backup-stream/src/utils.rs]: done

File: components/external_storage/export/src/export.rs (hunk 2 above)
Comment (Leavrth) [reply components/external_storage/export/src/export.rs]: done

File: components/external_storage/export/src/export.rs (hunk 6 above)
Comment (Leavrth) [reply components/external_storage/export/src/export.rs]: done

File: components/external_storage/src/lib.rs [hunk 7]
```diff
@@ -85,17 +116,25 @@ pub trait ExternalStorage: 'static + Send + Sync {
         &self,
         storage_name: &str,
         restore_name: std::path::PathBuf,
-        compressed_range: Option<(u64, u64)>,
         expected_length: u64,
-        expected_sha256: Option<Vec<u8>>,
         speed_limiter: &Limiter,
         file_crypter: Option<FileEncryptionInfo>,
```
Comment (YuJuncen) [inline components/external_storage/src/lib.rs:121]: Can we move `file_crypter` into `Options` too? So the `Options` fields contains all optional arguments of `restore`.

**Review by YuJuncen — APPROVED** [review]
rest lgtm

File: components/external_storage/src/lib.rs (hunk 7 above)
Comment (Leavrth) [reply components/external_storage/src/lib.rs]: done

**Review by overvenus — APPROVED** [review]

**Pull request merged**